File: tools/sms/src/main/java/top/bettercode/summer/tools/sms/SmsTemplate.java
package top.bettercode.summer.tools.sms;

import kotlin.jvm.functions.Function1;
import org.slf4j.Marker;
import org.slf4j.MarkerFactory;
import top.bettercode.summer.web.support.client.ApiTemplate;

/**
 * @author Peter Wu
 */
public abstract class SmsTemplate extends ApiTemplate {

  protected static final String LOG_MARKER_STR = "sms";
  public static final Marker LOG_MARKER = MarkerFactory.getMarker(LOG_MARKER_STR);

  public SmsTemplate(String collectionName, String name, String logMarker, int connectTimeout,
      int readTimeout) {
    super(collectionName, name, logMarker, connectTimeout, readTimeout, null, null);
  }

  public SmsTemplate(String collectionName, String name, String logMarker, int connectTimeout,
      int readTimeout, Function1<byte[], byte[]> requestDecrypt,
      Function1<byte[], byte[]> responseDecrypt) {
    super(collectionName, name, logMarker, connectTimeout, readTimeout, requestDecrypt,
        responseDecrypt);
  }
}

File: clients/sms/src/main/kotlin/top/bettercode/summer/tools/sms/SmsTemplate.kt
package top.bettercode.summer.tools.sms

import org.slf4j.Marker
import org.slf4j.MarkerFactory
import top.bettercode.summer.logging.annotation.LogMarker
import top.bettercode.summer.tools.lang.client.ApiTemplate

/**
 * @author Peter Wu
 */
@LogMarker(SmsTemplate.LOG_MARKER_STR)
abstract class SmsTemplate : ApiTemplate {
    constructor(collectionName: String, name: String, logMarker: String, connectTimeout: Int,
                readTimeout: Int) : super(collectionName, name, logMarker, connectTimeout, readTimeout, null, null)

    constructor(collectionName: String, name: String, logMarker: String, connectTimeout: Int,
                readTimeout: Int, requestDecrypt: ((ByteArray) -> ByteArray)?,
                responseDecrypt: ((ByteArray) -> ByteArray)?) : super(collectionName, name, logMarker, connectTimeout, readTimeout, requestDecrypt,
            responseDecrypt)

    companion object {
        const val LOG_MARKER_STR = "sms"

        @JvmField
        val LOG_MARKER: Marker? = MarkerFactory.getMarker(LOG_MARKER_STR)
    }
}
